### PR TITLE
Optimized news loading and added test for content_index view

### LINKED ISSUE #3: Use proper joins to render the index page
# Problem

Loading data in the main view of the `content` app doesn't use joins. That's OK until we have few news, but doesn't scale.

This is how news are loaded in the `index` view:

https://github.com/vrde/letscheck/blob/5348d08a58a25403f2af60695252ef1a1339fde8/src/content/views.py#L13-L17

And this is where the news are rendered:

https://github.com/vrde/letscheck/blob/5348d08a58a25403f2af60695252ef1a1339fde8/src/content/templates/content/index.html#L84-L107

This doesn't scale because for each news we are hitting the db again to load derived data.

# Solution

Use [`select_related`](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#select-related) or [`prefetch_related`](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#prefetch-related) to do the actual joins and load all data in one or two round-trips to the database.

# Deliverable

A PR that fixes the issue and test that verifies that the news are correctly rendered in the template. A similar test that can be used as inspiration is the one used for the `triage` app.

https://github.com/vrde/letscheck/blob/5348d08a58a25403f2af60695252ef1a1339fde8/src/triage/tests.py#L66-L110

### PR DESCRIPTION
**Pull request by lorrandal**
Fix #3  

I'm joining news only on rating `select_related('rating')`, I'm not super sure how to deal with the `Case` model but I can look into it!

### CODE REVIEW

File: src/content/tests.py [hunk 1]
```diff
@@ -1,3 +1,30 @@
...
+    def setUp(self):
+        self.c = Client()
+        self.c.force_login(User.objects.get_or_create(username="testuser")[0])
+
+    def test_index_should_display_news(self):
+        from .views import index
+        from triage.models import News, Rating
+
+        rating = Rating(name="rating1")
+        rating.save()
```
Comment (vrde) [inline src/content/tests.py]: To create and save an object in a single step, you can use the [create() method](https://docs.djangoproject.com/en/3.0/topics/db/queries/#creating-objects): `Rating.objects.create(...)`

File: src/content/tests.py [hunk 2]
```diff
@@ -1,3 +1,30 @@
...
+
+        news = News(
+            title="title1",
+            description="description1",
+            rating=rating)
+        news.save()
```
Comment (vrde) [inline src/content/tests.py]: Same here, you can use `News.objects.create(...)`.

File: src/content/tests.py [hunk 3]
```diff
@@ -1,3 +1,30 @@
...
+from django.urls import reverse
+from django.contrib.auth.models import User
+from django.test import Client, TestCase
 
-# Create your tests here.
+
+class ContentViewTest(TestCase):
+    def setUp(self):
+        self.c = Client()
+        self.c.force_login(User.objects.get_or_create(username="testuser")[0])
```
Comment (vrde) [inline src/content/tests.py]: For the index page we don't need to have a logged in user, you can remove this part.

**Review by vrde — CHANGES_REQUESTED** [review]
Great job! I added a couple of suggestions to simplify the code.

About the `create` method, it's not used everywhere in the code but I'd like to have it as the standard approach when creating instances of models.

File: src/content/tests.py (hunk 3 above)
Comment (lorrandal) [reply src/content/tests.py]: thanks for the inputs, I followed up with a commit

**Review by vrde — APPROVED** [review]

**Pull request merged**